### PR TITLE
#DiceGame Still there are bugs 

### PR DESCRIPTION
**Pull request by mafromist**
but the game is a bit working now

### CODE REVIEW

File: app2.js [hunk 1]
```diff
@@ -1,53 +1,48 @@
...
+
+const rollBtn = document.querySelector("#rollBtn");
+const holdBtn = document.querySelector("#holdBtn");
+
+
 const newGame = () => {
-    console.log("newGame");
-    
+    pickAPlayer();
+    currentPlayer = pickAPlayer();
```
Comment (nmunro) [inline app2.js:35]: Is line 34 needed?

File: app2.js [hunk 2]
```diff
@@ -60,79 +55,99 @@ newGameBtn.addEventListener("click", () => {
...
+        if (currentPlayer == 1) {
+            player1Score = player1Score + currentDice;
+            currentScore1.textContent = player1Score;
+        } else {
+            player2Score = player2Score + currentDice;
+            currentScore2.textContent = player2Score;
+        }
+    } else if (currentDice !== 1 && hold1Score !== 0) {    
+        if (currentPlayer == 1) {
+            let afterHoldScore1 = 0;
```
Comment (nmunro) [inline app2.js:77]: Is this used anywhere?

File: app2.js [hunk 3]
```diff
@@ -60,79 +55,99 @@ newGameBtn.addEventListener("click", () => {
...
+            player1Score = hold1Score + currentDice ;
+            currentScore1.textContent = player1Score;
+        } else {
+            player2Score = hold2Score + currentDice;
+            currentScore2.textContent = player2Score;
+        }
+    } else if (currentDice === 1) {
+        currentScore1.textContent = (currentPlayer === 1) ? "You Lost" : "You Win!";
+        currentScore2.textContent = (currentPlayer === 2) ? "You Lost" : "You Win!";
```
Comment (nmunro) [inline app2.js:86]: Nice 👍

File: app2.js [hunk 4]
```diff
@@ -60,79 +55,99 @@ newGameBtn.addEventListener("click", () => {
...
         currentScore1.textContent = 0;
-   
+        p1Bg.style.backgroundColor = "transparent";
+        p2Bg.style.backgroundColor = "white";
     } else {
+        hold2Score = player2Score;
         holdScore2.textContent = player2Score;
         currentScore2.textContent = 0;
+        p1Bg.style.backgroundColor = "transparent";
+        p2Bg.style.backgroundColor = "white";
```
Comment (nmunro) [inline app2.js:110]: This is the same as 103 & 104, is this a bug or just duplicated code that can be moved?

File: app2.js [hunk 5]
```diff
@@ -60,79 +55,99 @@ newGameBtn.addEventListener("click", () => {
...
 
+const  switchBg = () => { 
+    if (currentPlayer === 1) {
+        p1Bg.style.backgroundColor = "white";
+        p2Bg.style.backgroundColor = "transparent";
+    } else {
+        p1Bg.style.backgroundColor = "transparent";
+        p2Bg.style.backgroundColor = "white";
+    }
+}
```
Comment (nmunro) [inline app2.js:134]: In fact doesn't this do what 103, 104, 109 and 110 do? Looks like you can save a few lines of code?

**Review by nmunro — COMMENTED** [review]
Really like the way you're looking at splitting code up into small dedicated functions!

File: app2.js (hunk 1 above)
Comment (mafromist) [reply app2.js:35]: yeah no need for it 👍  thank you :)

File: app2.js (hunk 4 above)
Comment (mafromist) [reply app2.js:110]: Actually you are right I just deleted the 102,103,  109,110 and the function has same features of them

File: app2.js (hunk 5 above)
Comment (mafromist) [reply app2.js:134]: yeap i saved 4 lines :)

**Pull request merged**